File: server/humanai/src/main/java/gotrainer/humanai/MoveSuggestion.java
package gotrainer.humanai;

public class MoveSuggestion {

    public Move move;
    public int visits;
    public float winrate;
    public float scoreLead;


    public MoveSuggestion() {}

    public MoveSuggestion(String color, String coord, String visits, String winrate, String scoreLead) {
        setMove(color, coord);
        setVisits(visits);
        setWinrate(winrate);
        setScoreLead(scoreLead);
    }

    public void setMove(String color, String coord) {
        this.move = new Move(color, coord);
    }

    public void setVisits(String visits) {
        this.visits = Integer.parseInt(visits);
    }

    public void setWinrate(String winrate) {
        this.winrate = Math.round(Float.parseFloat(winrate) * 10000) / 100f;
    }

    public void setScoreLead(String scoreLead) {
        this.scoreLead = Math.round(Float.parseFloat(scoreLead) * 10) / 10f;
    }

}

File: server/TempAPI/TempAPI/KataGo/MoveSuggestion.cs
namespace TempAPI.KataGo
{
    public class MoveSuggestion
    {
        public Move move { get; set; }
        public int visits { get; set; }
        public float winrate { get; set; }
        public float scoreLead { get; set; }

        public MoveSuggestion() { }

        public MoveSuggestion(string color, string coord, string visits, string winrate, string scoreLead)
        {
            SetMove(color, coord);
            SetVisits(visits);
            SetWinrate(winrate);
            SetScoreLead(scoreLead);
        }

        public MoveSuggestion(string color, string coord, int visits, float winrate, float scoreLead)
        {
            SetMove(color, coord);
            this.visits = visits;
            this.winrate = winrate;
            this.scoreLead = scoreLead;
        }

        public void SetMove(string color, string coord)
        {
            move = new Move(color, coord);
        }

        public void SetVisits(string visits)
        {
            this.visits = int.Parse(visits);
        }

        public void SetWinrate(string winrate)
        {
            this.winrate = (float)(Math.Round(float.Parse(winrate) * 10_000.0) / 100.0);
        }

        public void SetScoreLead(string scoreLead)
        {
            this.scoreLead = (float)(Math.Round(float.Parse(scoreLead) * 10.0) / 10.0);
        }
    }
}
